app/src/main/java/org/simple/clinic/enterotp/EnterOtpScreenController.kt

```diff
@@ -97,4 +98,33 @@ class EnterOtpScreenController @Inject constructor(private val userSession: User
         .compose(NewlyVerifiedUser())
         .map { { ui: Ui -> ui.goBack() } }
   }
+
+  private fun resendSms(events: Observable<UiEvent>): Observable<UiChange> {
+
+    return events.ofType<EnterOtpResendSmsClicked>()
+        .flatMap { _ ->
+          userSession
+              .requestLoginOtp()
+              .flatMapObservable { loginResult ->
+                Observable.merge(handleRequestOtpResult(loginResult), hideProgressOnLoginResult(loginResult))
+              }
+              .startWith { ui: Ui ->
+                ui.hideError()
+                ui.showProgress()
+              }
+        }
+  }
+
+  private fun handleRequestOtpResult(loginResult: LoginResult): Observable<UiChange> {
```

Comment:
Please don't use `handle` for naming functions. All functions are anyway handling something. :)

Reply:
Renamed to `showMessageOnLoginResult`.